File: travelsathi-backend/aiservice/src/main/java/com/travelsathi/aiservice/models/Trip.java
package com.travelsathi.aiservice.models;

import java.time.LocalDate;
import java.time.LocalDateTime;

import lombok.Data;

@Data
public class Trip {
    private String id;
    private String userId;
    private String source;
    private String destination;
    private Integer numberOfTravelers;
    private String additionalPreference;
    private LocalDate startDate;
    private LocalDate endDate;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: frontend-app/travelSathi/src/types/Trip.ts
/**
 * This interface represents the Trip object in the Backend
 */
export interface Trip {
    id: string;
    userId: string,
    source: string;
    destination: string;
    startDate: string;
    endDate: string;
    numberOfTravelers: number;
    additionalPreference: string;
    createdAt: string;
    updatedAt: string;
}